Source/Core/VideoCommon/VertexLoaderARM64.cpp

```diff
@@ -420,9 +434,11 @@ void VertexLoaderARM64::GenerateVertexLoader()
     STR(IndexType::Unsigned, scratch1_reg, dst_reg, m_dst_ofs);
 
     // Z-Freeze
-    CMP(count_reg, 3);
-    FixupBranch dont_store = B(CC_GT);
-    MOVP2R(EncodeRegTo64(scratch2_reg), VertexLoaderManager::position_matrix_index);
+    CMP(remaining_reg, 3);
+    FixupBranch dont_store = B(CC_GE);
+    MOVP2R(EncodeRegTo64(scratch2_reg), VertexLoaderManager::position_matrix_index_cache.data());
+    ADD(EncodeRegTo64(scratch2_reg), EncodeRegTo64(scratch2_reg), EncodeRegTo64(remaining_reg),
+        ArithOption(EncodeRegTo64(remaining_reg), ShiftType::LSL, 2));
     STR(IndexType::Unsigned, scratch1_reg, EncodeRegTo64(scratch2_reg), 0);
```

Comment:
Optimized version using an index register:

```suggestion
    STR(IndexType::Unsigned, scratch1_reg, EncodeRegTo64(scratch2_reg), ArithOption(remaining_reg, true));
```

Reply:
The `IndexType::Unsigned` argument can't be included in that case, so I've changed it to this:

``C++
STR(scratch1_reg, EncodeRegTo64(scratch2_reg), ArithOption(remaining_reg, true));
```

(I don't have easy access to an ARM device to test on, so I don't know if this works.)

Reply:
Ah yes, that is correct. I must admit that I haven't tried compiling the code I'm giving you :)

Once you feel the code is mature enough, feel free to ask me to test the whole PR on AArch64. I have RS3 but not RS2.